File: datastage-server/src/utils/logger.ts
/**
 * Simple logging utility
 */

import { ENV } from '../config/constants.js';

type LogLevel = 'debug' | 'info' | 'warn' | 'error';

class Logger {
  private level: LogLevel;

  constructor() {
    this.level = (ENV.LOG_LEVEL as LogLevel) || 'info';
  }

  private shouldLog(level: LogLevel): boolean {
    const levels: LogLevel[] = ['debug', 'info', 'warn', 'error'];
    const currentLevelIndex = levels.indexOf(this.level);
    const messageLevelIndex = levels.indexOf(level);
    return messageLevelIndex >= currentLevelIndex;
  }

  private formatMessage(level: LogLevel, message: string, meta?: any): string {
    const timestamp = new Date().toISOString();
    const metaStr = meta ? ` ${JSON.stringify(meta)}` : '';
    return `[${timestamp}] [${level.toUpperCase()}] ${message}${metaStr}`;
  }

  debug(message: string, meta?: any): void {
    if (this.shouldLog('debug')) {
      console.error(this.formatMessage('debug', message, meta));
    }
  }

  info(message: string, meta?: any): void {
    if (this.shouldLog('info')) {
      console.error(this.formatMessage('info', message, meta));
    }
  }

  warn(message: string, meta?: any): void {
    if (this.shouldLog('warn')) {
      console.error(this.formatMessage('warn', message, meta));
    }
  }

  error(message: string, meta?: any): void {
    if (this.shouldLog('error')) {
      console.error(this.formatMessage('error', message, meta));
    }
  }
}

export const logger = new Logger();

// Made with Bob

File: python-mcp/src/utils/logger.py
"""
Simple logging utility
"""

import sys
import json
from datetime import datetime
from typing import Any, Optional
from ..config.constants import ENV


class Logger:
    """Simple logger for DataStage MCP Server"""
    
    def __init__(self):
        self.level = ENV.LOG_LEVEL.lower()
        self.levels = ['debug', 'info', 'warn', 'error']
    
    def _should_log(self, level: str) -> bool:
        """Check if message should be logged based on current log level"""
        try:
            current_level_index = self.levels.index(self.level)
            message_level_index = self.levels.index(level)
            return message_level_index >= current_level_index
        except ValueError:
            return True
    
    def _format_message(self, level: str, message: str, meta: Optional[Any] = None) -> str:
        """Format log message with timestamp and metadata"""
        timestamp = datetime.utcnow().isoformat() + 'Z'
        meta_str = f' {json.dumps(meta)}' if meta else ''
        return f'[{timestamp}] [{level.upper()}] {message}{meta_str}'
    
    def debug(self, message: str, meta: Optional[Any] = None) -> None:
        """Log debug message"""
        if self._should_log('debug'):
            print(self._format_message('debug', message, meta), file=sys.stderr)
    
    def info(self, message: str, meta: Optional[Any] = None) -> None:
        """Log info message"""
        if self._should_log('info'):
            print(self._format_message('info', message, meta), file=sys.stderr)
    
    def warn(self, message: str, meta: Optional[Any] = None) -> None:
        """Log warning message"""
        if self._should_log('warn'):
            print(self._format_message('warn', message, meta), file=sys.stderr)
    
    def error(self, message: str, meta: Optional[Any] = None) -> None:
        """Log error message"""
        if self._should_log('error'):
            print(self._format_message('error', message, meta), file=sys.stderr)


# Global logger instance
logger = Logger()


# Made with Bob